trk/src/Screens/TabScreens/GymAnalytics/GymDaily/Backend/analyticsCalculations.js

```diff
@@ -66,17 +70,34 @@ export const fetchTapsAndCalculateAscents = async (yourClimbs) => {
   let peakTimeString = peakHour ? `${peakHour}:00 - ${parseInt(peakHour) + 1}:00` : '';
 
   // Determine most and least completed climbs
-  let mostCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count > b.count) ? a : b, null);
-  let leastCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count < b.count) ? a : b, null);
-
+  let mostCompletedClimbs = [];
+  let leastCompletedClimbs = [];
+  let maxCount = 0;
+  let minCount = Infinity;
+  
+  Object.values(climbFrequencyMap).forEach(climb => {
+    if (climb.count > maxCount) {
+      mostCompletedClimbs = [climb];
+      maxCount = climb.count;
+    } else if (climb.count === maxCount) {
+      mostCompletedClimbs.push(climb);
+    }
+  
+    if (climb.count < minCount) {
+      leastCompletedClimbs = [climb];
+      minCount = climb.count;
+    } else if (climb.count === minCount) {
+      leastCompletedClimbs.push(climb);
+    }
+  });
   // Return the calculated values
   return {
```

Comment:
most and least completed climb are now arrays. the logic behind maxCount and minCount is:
-  maxCount set to 0 so any climb with higher count becomes highest tap count as we iterate through the climb array
- minCount set to infinity so any climb with lower count becomes lowest tap count as we iterate through array

Reply:
Looks perfect!